crates/bevy_core/Cargo.toml

```diff
@@ -17,6 +17,8 @@ bevy_math = { path = "../bevy_math", version = "0.9.0-dev" }
 bevy_reflect = { path = "../bevy_reflect", version = "0.9.0-dev", features = ["bevy"] }
 bevy_tasks = { path = "../bevy_tasks", version = "0.9.0-dev" }
 bevy_utils = { path = "../bevy_utils", version = "0.9.0-dev" }
+bevy_window = { path = "../bevy_window", version = "0.9.0-dev" }
+bevy_input = { path = "../bevy_input", version = "0.9.0-dev" }
```

Comment:
IMO we at least need to make this dependency optional, and block it behind a feature flag that's on by default. Why are you opposed?

Reply:
To be honest, I'd not looked into `bevy_core` before today.

My understanding is that its purpose is glue code for the lowest levels/things which every bevy application needs. 

Being able to close a window nicely is not core functionality - e.g. it's meaningless for headless apps.

Perhaps we should uplift every event type which only makes sense in terms of windows existing into `bevy_window`. That to me seems a more logical split.

To be honest, having looked at bevy_core, everything there except `Name` should be in `bevy_app` and the `App::new()` constructor instead. As far as I can tell, [`register_type`](https://docs.rs/bevy/latest/bevy/app/struct.App.html#method.register_type) is already in `bevy_app`. And `DefaultTaskPoolOptions` also seems very 'appy' to me, since afaik `bevy_app` relies on it.

Reply:
Yeah, I'd honestly like most of the contents of `bevy_core` to live in a higher level `bevy_helpers` crate.

> Perhaps we should uplift every event type which only makes sense in terms of windows existing into bevy_window. That to me seems a more logical split.

I'd be fine with this, but I think it's reasonable for users to want to find all of the input events in `bevy_input`.

Reply:
> I'd be fine with this, but I think it's reasonable for users to want to find all of the input events in bevy_input.

Perhaps we could move `Touch`(es) into `bevy_window` and then re-export it from `bevy_input`, provided `core::close_on_esc` stays in `bevy_core` or some `bevy_helpers` crate.